torch/testing/asserts.py

```diff
@@ -0,0 +1,186 @@
+
+    if tensor1.device != tensor2.device:
+        tensor1 = tensor1.cpu()
+        tensor2 = tensor2.cpu()
+
+    if tensor1.dtype != tensor2.dtype:
+        dtype = _get_comparison_dtype(tensor1, tensor2)
+        tensor1 = tensor1.to(dtype)
+        tensor2 = tensor2.to(dtype)
+
+    asserter(tensor1, tensor2)
+
+
+def _tensors_equal_asserter(tensor1: torch.Tensor, tensor2: torch.Tensor):
+    if not torch.all(torch.eq(tensor1, tensor2)):
+        raise AssertionError("ADDME")
+
+
+@_hide_internal_traceback
+def assert_tensors_equal(
+    tensor1: torch.Tensor, tensor2: torch.Tensor, *, warn_floating_point: bool = True, **strict_attributes: bool
+) -> None:
+    """Assert that two tensors are equal.
+
+    Before the values are compared, the ``shape`` and the ``dtype`` of the tensors are checked for equality. You can
+    change this behavior with the ``strict_attributes`` parameter.
+
+    Depending on the strictness some additional steps may be performed before the value comparison:
+
+    - If the tensors don't live in the same device memory they are moved to CPU memory.
+    - If the tensors don't have the same ``dtype``, they are converted to the one determined by
+        :func:`torch.promote_types`.
+
+    Args:
+        tensor1 (torch.Tensor): First tensor.
+        tensor2 (torch.Tensor): Second tensor.
+        warn_floating_point (bool): Emit a warning if at least one of the inputs is a floating point tensor.
+        **strict_attributes (bool): Set strictness for attribute equality checking. If omitted, this defaults to
+            ``shape=True, dtype=True``.
+
+    Raises:
+        AssertionError: If the specified attributes do not match.
+        AssertionError: If the tensors are not equal.
+    """
+    is_floating_point = tensor1.dtype.is_floating_point or tensor2.dtype.is_floating_point
+    if is_floating_point and warn_floating_point:
+        warnings.warn(
+            "Due to the limitations of floating point arithmetic, comparing floating-point tensors for equality is not "
+            "recommended. Use the respective 'allclose' variant instead.",
+            UsageWarning,
+        )
+
+    _compare_tensors(tensor1, tensor2, _tensors_equal_asserter, **strict_attributes)
+
+
+def _tensors_allclose_asserter(
+    tensor1: torch.Tensor, tensor2: torch.Tensor, rtol: Optional[float], atol: Optional[float]
+) -> None:
+    default_rtol, default_atol = _get_default_rtol_and_atol(tensor1, tensor2)
+    rtol = rtol if rtol is not None else default_rtol
+    atol = atol if atol is not None else default_atol
+
+    if not torch.allclose(tensor1, tensor2, rtol=rtol, atol=atol):
+        raise AssertionError("ADDME")
+
+
+@_hide_internal_traceback
+def assert_tensors_allclose(
+    tensor1: torch.Tensor,
+    tensor2: torch.Tensor,
+    *,
+    rtol: Optional[float] = None,
+    atol: Optional[float] = None,
+    **strict_attributes: bool,
+) -> None:
+    """Assert that two tensors are equal up to a desired precision.
```

Comment:
"Assert" -> "Asserts"

Comments and documentation describe what code or operations do. If we think about it from that perspective, we can think about a sentence like, "assert_tensors_allclose() asserts that two..." which sounds more natural than "assert_tensors_allclose() assert that two...".

Reply:
As above: what does "equal" mean?